### PR TITLE
🎨 Generate theme from a primary color

### PR DESCRIPTION
**Pull request by hbernardi-luna**
## 📜 Description
With https://github.com/material-foundation/material-color-utilities code, make two utilities method to build a dynamic theme depending on user color choice.

Documentation available [here](https://www.notion.so/lunabeestudio/Lunabee-Compose-library-42f4d91d463a4e8492a1a4d3d32a243f#3508210f150a4dafb0c70b84bc4ba1e2)

## 💡 Motivation and Context
App customization

## 💚 How did you test it?

https://user-images.githubusercontent.com/72007948/197217635-f9b9056c-76d9-4043-95db-10f928cf01a2.mov

+ compare value generated on Material site https://m3.material.io/theme-builder#/custom

## 📝 Checklist
* [x] I compiled before submitting the PR
* [x] I reviewed the submitted code
* [x] I launched `./gradlew detekt`

### CODE REVIEW

**Comment by hbernardi-luna** [comment]
@qmetzler-luna I just made a fix for publication (material-color-utilities was not packaged)

File: app/src/main/java/studio/lunabee/compose/material3/theme/ThemeScreen.kt [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+        Text(
+            text = color.hexValue,
+            style = MaterialTheme.typography.bodySmall,
+        )
+    }
+}
+
+val Color.hexValue: String
+    get() = String.format(locale = Locale.getDefault(), format = "#%08X", toArgb())
```
Comment (qmetzler-luna) [inline app/src/main/java/studio/lunabee/compose/material3/theme/ThemeScreen.kt:219]:
```suggestion
    get() = String.format(locale = Locale.ROOT, format = "#%08X", toArgb())
```

**Review by qmetzler-luna — COMMENTED** [review]
- Didn't read the `.java` code, don't know if I should have 🙈
- I don't understand what's the colour at the right of the input. is it the primary? Why is it different that the one entered? 🤷‍♂️

**Comment by hbernardi-luna** [comment]
> * Didn't read the `.java` code, don't know if I should have 🙈
> * I don't understand what's the colour at the right of the input. is it the primary? Why is it different that the one entered? 🤷‍♂️

@qmetzler-luna 
1. Well, I think it's not your job to review Material foundation code :p I didn't modified it, I just removed useless file.
2. It's the primary generated based on the color you entered. It seems that Material Design change it a little bit in order to make a beautiful palette. Maybe I can add an explanation. On [Material Design](https://m3.material.io/theme-builder#/custom) site, you can observe that putting #FF00FF in primary will generate for example #a900a9, or if you put #FFFFFF, your primary color will not be white at all. Maybe I can add a small text of explanation.

**Comment by qmetzler-luna** [comment]
> Maybe I can add a small text of explanation.

That would be nice 🤗

**Review by qmetzler-luna — APPROVED** [review]

**Pull request merged**